plugins/scaffolder-react/src/hooks/useEventStream.ts

```diff
@@ -169,67 +169,82 @@ export const useTaskEventStream = (taskId: string): TaskStream => {
+          if (didCancel) {
+            return;
           }
-        }
+          dispatch({ type: 'INIT', data: task });
 
-        logPusher = setInterval(emitLogs, 500);
-
-        subscription = observable.subscribe({
-          next: event => {
-            switch (event.type) {
-              case 'log':
-                return collectedLogEvents.push(event);
-              case 'cancelled':
-                dispatch({ type: 'CANCELLED' });
-                return undefined;
-              case 'completion':
-                emitLogs();
-                dispatch({ type: 'COMPLETED', data: event });
-                return undefined;
-              default:
-                throw new Error(
-                  `Unhandled event type ${event.type} in observer`,
-                );
+          // TODO(blam): Use a normal fetch to fetch the current log for the event stream
+          // and use that for an INIT_EVENTs dispatch event, and then
+          // use the last event ID to subscribe using after option to
+          // stream logs. Without this, if you have a lot of logs, it can look like the
+          // task is being rebuilt on load as it progresses through the steps at a slower
+          // rate whilst it builds the status from the event logs
+          const observable = scaffolderApi.streamLogs({ taskId });
+
+          const collectedLogEvents = new Array<LogEvent>();
+
+          function emitLogs() {
+            if (collectedLogEvents.length) {
+              const logs = collectedLogEvents.splice(
+                0,
+                collectedLogEvents.length,
+              );
+              dispatch({ type: 'LOGS', data: logs });
             }
-          },
-          error: error => {
-            emitLogs();
+          }
+
+          logPusher = setInterval(emitLogs, 500);
+
+          subscription = observable.subscribe({
+            next: event => {
+              switch (event.type) {
+                case 'log':
+                  return collectedLogEvents.push(event);
+                case 'cancelled':
+                  dispatch({ type: 'CANCELLED' });
+                  return undefined;
+                case 'completion':
+                  emitLogs();
+                  dispatch({ type: 'COMPLETED', data: event });
+                  return undefined;
+                default:
+                  throw new Error(
+                    `Unhandled event type ${event.type} in observer`,
+                  );
+              }
+            },
+            error: error => {
+              emitLogs();
+              // in some cases the error is a refused connection from backend
+              // this can happen from internet issues or proxy problems
+              // so we try to reconnect again after some time
+              // just to restart the fetch process
+              // details here https://github.com/backstage/backstage/issues/15002
+
+              if (!error.message) {
+                error.message =
```

Comment:
Interestingly what is the error type at this point and why does it not have a message?

Reply:
I just was getting an `ERR_NETWORK_CHANGED 200` in the browser, but didn't find a simple way to catch this error in the code, so I used the empty message to change this.

Do you have suggestions on how to catch this?

![image](https://github.com/backstage/backstage/assets/1574240/4ba2ee9c-2557-41d0-a056-49c018eeeb7f)

Reply:
Hmm - I wonder if these are just browser errors and are not available in javascript though. If you stringify the error that you get in the `useEventStream` and log it what does it look like?

Reply:
It is an [event](https://developer.mozilla.org/en-US/docs/Web/API/EventSource/error_event) from the EventSource, so just generic information 🤔 
![image](https://github.com/backstage/backstage/assets/1574240/8a1f37a5-536a-49ea-8e6b-995eee77c001)